### PR TITLE
template: GetTemplateByName and sorting for []*Template

### PR DESCRIPTION
**Pull request by jessicatoscani**
```
go test -timeout 30s -testify.m ^(TestClient_GetTemplateByName)$ github.com/exoscale/egoscale/v2

PASS
ok  	github.com/exoscale/egoscale/v2	3.015s

```

### CODE REVIEW

File: v2/template.go [hunk 1]
```diff
@@ -29,6 +30,22 @@ type Template struct {
 	Zone            *string
 }
 
+// Implement sort.Interface for []*Template by CreatedAt or by Nane
+type Templates []*Template
+
+func (s Templates) Len() int      { return len(s) }
+func (s Templates) Swap(i, j int) { s[i], s[j] = s[j], s[i] }
```
Comment (greut) [inline v2/template.go:37]: Just use generics, https://pkg.go.dev/golang.org/x/exp/slices#Sort

**Review by arapaho — APPROVED** [review]

File: v2/template.go (hunk 1 above)
Comment (PhilippeChepy) [reply v2/template.go:37]: This project is still buildable with go 1.17 for now; generics are available from 1.18.x

**Review by PhilippeChepy — APPROVED** [review]

File: v2/template.go (hunk 1 above)
Comment (greut) [reply v2/template.go:37]: 1.20 is around the corner. It's likely you won't lose anything by bumping it to 1.18.

Comment (jessicatoscani) [reply v2/template.go:37]: We will do it later

**Pull request merged**